eoc_journal/eoc_journal.py

```diff
@@ -84,6 +90,13 @@ def student_view(self, context=None):
         """
         context = context.copy() if context else {}
 
+        user = self._get_current_user()
+        course_id = getattr(self.runtime, 'course_id', 'course_id')
+        course_id = unicode(_normalize_id(course_id))
+        anonymous_user_id = anonymous_id_for_user(user, course_id, save=False)
```

Comment:
As mentioned above, we should duplicate [the problem builder approach](https://github.com/open-craft/problem-builder/blob/27e2f4ba5afdefcb6a4e5fd793969e1e626a0abe/problem_builder/answer.py#L60-L65) instead of using the function from edx-platform.